File: sdk/src/test/java/com/adadapted/android/sdk/ext/json/JsonSessionRequestBuilderTest.java
package com.adadapted.android.sdk.ext.json;

import com.adadapted.android.sdk.core.device.model.DeviceInfo;
import com.adadapted.android.sdk.core.session.SessionRequestBuilder;

import junit.framework.Assert;

import org.json.JSONObject;
import org.junit.Test;

/**
 * Created by chrisweeden on 4/6/15.
 */
public class JsonSessionRequestBuilderTest {
    private static final String APPID = "TESTAPPLICATION";
    private static final String UDID = "TESTUDID";
    private static final String BUNDLEID = "TESTBUNDLEID";
    private static final String[] ZONES = new String[]{"10", "11"};
    private static final String DEVICE = "TESTDEVICE";
    private static final String OS = "TESTOS";
    private static final String OSV = "TESTOSV";
    private static final String LOCALE = "locale";
    private static final String TIMEZONE = "timezone";
    private static final int DH = 600;
    private static final int DW = 320;
    private static final int ALLOWRETARGETING = 1;
    private static final String SDKVERSION = "TESTSDKVERSION";

    @Test
    public void testBuildSessionRequestJson_ReturnsJson() throws Exception {
        DeviceInfo deviceInfo = new DeviceInfo();
        deviceInfo.setAppId(APPID);
        deviceInfo.setZones(ZONES);
        deviceInfo.setUdid(UDID);
        deviceInfo.setBundleId(BUNDLEID);
        deviceInfo.setDevice(DEVICE);
        deviceInfo.setOs(OS);
        deviceInfo.setOsv(OSV);
        deviceInfo.setDh(DH);
        deviceInfo.setDw(DW);
        deviceInfo.setSdkVersion(SDKVERSION);

        SessionRequestBuilder builder = new JsonSessionRequestBuilder();
        JSONObject json = builder.buildSessionInitRequest(deviceInfo);

        Assert.assertEquals(APPID, json.getString(JsonFields.APPID));
    }
}

File: advertising_sdk/src/test/java/com/adadapted/android/sdk/ext/json/JsonSessionRequestBuilderTest.kt
package com.adadapted.android.sdk.ext.json

import com.adadapted.android.sdk.core.http.HttpAppEventSinkTest
import org.junit.Assert.assertEquals
import org.junit.Test

class JsonSessionRequestBuilderTest {

    @Test
    fun buildSessionInitRequest() {
        val mockDeviceInfo = HttpAppEventSinkTest.generateMockDeviceInfo()
        val testJsonSessionRequestBuilder = JsonSessionRequestBuilder()

        val resultRequest = testJsonSessionRequestBuilder.buildSessionInitRequest(mockDeviceInfo)
        assertEquals("testUdid", resultRequest.get("udid"))
        assertEquals(true, resultRequest.get("allow_retargeting"))
        assertEquals("testAppId", resultRequest.get("app_id"))
        assertEquals("testBundleId", resultRequest.get("bundle_id"))
        assertEquals("testBundleVersion", resultRequest.get("bundle_version"))
        assertEquals("testDevice", resultRequest.get("device_name"))
        assertEquals("DeviceUdId", resultRequest.get("device_udid"))
        assertEquals("Android", resultRequest.get("device_os"))
        assertEquals("testOsv", resultRequest.get("device_osv"))
        assertEquals("testLocale", resultRequest.get("device_locale"))
        assertEquals("testTimeZone", resultRequest.get("device_timezone"))
        assertEquals("testCarrier", resultRequest.get("device_carrier"))
        assertEquals(0, resultRequest.get("device_height"))
        assertEquals(0, resultRequest.get("device_width"))
        assertEquals("0", resultRequest.get("device_density"))
        assertEquals("testLocale", resultRequest.get("device_locale"))
    }
}